app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -57,7 +62,14 @@ class AccountController @Inject()(
       _ <- businessEntityJourneyOpt.fold(store.put(BusinessEntityJourney.fromUndertakingOpt(retrievedUndertaking)))(Future.successful)
       _ <- if (retrievedUndertaking.isDefined) store.put(retrievedUndertaking.getOrElse(sys.error("Undertaking is Missing"))) else Future.successful(Unit)
     } yield (retrievedUndertaking, eligibilityJourney, undertakingJourney) match {
-      case (Some(undertaking), _, _) => Ok(accountPage(undertaking, !undertaking.getAllNonLeadEORIs().isEmpty))
+
+      case (Some(undertaking), _, _) =>
+
+        val lastDayToReportDate = undertaking.lastSubsidyUsageUpdt.map(_.plusDays(REPORT_DEMINIMIS_DUE_DAY))
```

Comment:
NABD - but this could be in ReportDeMinimisReminderHelper - I don't think the controller needs to know about the 90 day max.

Reply:
Yeah I tend to prefer pulling out logic like this so nicely encapsulated and easy to test.